sdks/java/core/src/main/java/org/apache/beam/sdk/io/TextIO.java

```diff
@@ -745,8 +761,10 @@ public PDone apply(PCollection<T> input) {
       public void populateDisplayData(DisplayData.Builder builder) {
         super.populateDisplayData(builder);
 
+        String prefixString = filenamePrefix.isAccessible()
+            ? filenamePrefix.get() : filenamePrefix.toString();
         builder
-            .addIfNotNull(DisplayData.item("filePrefix", filenamePrefix)
+            .addIfNotNull(DisplayData.item("filePrefix", prefixString)
```

Comment:
I wonder if `DisplayData` itself should be made `ValueProvider` aware. This logic is probably exactly the same frickin' everywhere. @swegner @bjchambers ?